components/support/rc_crypto/src/hkdf.rs

```diff
@@ -0,0 +1,99 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use crate::{
+    digest,
+    error::*,
+    hmac, p11,
+    util::{ensure_nss_initialized, map_nss_secstatus},
+};
+use nss_sys::*;
+use std::{mem, ops::Deref, os::raw::c_uchar, ptr};
+
+pub fn extract_and_expand(
+    salt: &hmac::SigningKey,
+    secret: &[u8],
```

Comment:
Huh. It seems weird to me that `salt` is a `SigningKey` while `secret` is raw bytes, but this matches what ring does so *shrug*...I guess it just need to be this way in order for the underlying `hmac` calls to work correctly?

Reply:
Correct, from ring doc:
```
Salts have type hmac::SigningKey instead of &[u8] because they are frequently used for multiple HKDF operations, and it is more efficient to construct the SigningKey once and reuse it. Given a digest algorithm digest_alg and a salt salt: &[u8], the SigningKey should be constructed as hmac::SigningKey::new(digest_alg, salt).
```